src/base/preferences.cpp

```diff
@@ -448,6 +448,26 @@ void Preferences::setActionOnDblClOnTorrentFn(int act)
     setValue("Preferences/Downloads/DblClOnTorFn", act);
 }
 
+bool Preferences::deleteAddedTorrents() const
```

Comment:
Well, maybe four static functions in the `TorrentFileProcessignGuard`  will do the trick, but the class interface will look not that clean.